### DISCUSSION TITLE
Add hero artifacts to the email notification

### DISCUSSION THREAD

**Discussion by hedrinjt** [HughJeffner/FFRK-LabMem#188] [Ideas]
I am trying to add some more detail to the email notification.  Is there a loot variable that is available to the EmailNotification class?  I would like to just receive in an email the Hero Artifacts that were obtained during the last run.  Or do I alter the class and send through that array to put it in the message or should I just attach the CSV log?

**Comment by HughJeffner** [comment]
I'm away from my workstation so this might not be exactly correct but you should be able to access it in the counters

`FFRK_LabMem.Data.Counters.Default.CounterSets["CurrentLab"].HeroEquipment`

**Reply by HughJeffner** [reply]
I looked and you won't be able to access those counters from the `EmailNotificaton` class directly

Easiest way would be to construct the email body in `Lab.Actions` file and pass it as the message to the `Notify` method: https://github.com/HughJeffner/FFRK-LabMem/blob/d6bce9172c01c77d59c03f64e76e6d217d31077d/FFRK-LabMem/Machines/Lab.Actions.cs#L727

I'll have to think of a more robust solution for this, perhaps add a Body field to the email settings and have replacement tokens or something.

**Reply by hedrinjt** [reply]
Is that accessible from there?  When I try to add `using FFRK_LabMem;` at the top of that .cs it has a red error under it.  I dabble but don't understand how to pull that information into this class so it is usable.  If I just try to assign it, I get a similar error.  How do I make a class under the FFRK_LabMem namespace accessible to FFRK_Machines.Services.Notifications?

**Reply by hedrinjt** [reply]
lol, typing at the same time.  I'll take a look

**Reply by hedrinjt** [reply]
That worked perfectly.  I just converted the SortedDictionary variable you have `FFRK_LabMem.Data.Counters.Default.CounterSets["CurrentLab"].HeroEquipment` to a string and now I get:

> LabMem event: LAB_COMPLETED
>
>[Cid's Finery (XIII), 1]
>[General's Belt (XIII), 1]
>[Muscle Belt (VI), 2]
>[Save the Queen (IX), 1]

Thank you for your replies!
